docs/fsharp/language-reference/compiler-messages/fs0008.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "Compiler Error FS0008"
+ms.date: 01/07/2020
+f1_keywords:
+  - "FS0008"
+helpviewer_keywords:
+  - "FS0008"
+---
+
+# FS0008: Indeterminate runtime coercion
+
+This message is given when a type-test pattern is used to match a value, but the cast cannot be safely proven at compile time because the type value has not been determined by the compiler. This often happens when matching on a function parameter whose type has been inferred.  The following code illustrates an example of this:
+
+[!code-fsharp[FS0008-error](~/samples/snippets/fsharp/compiler-messages/fs0008.fs#2-5)]
+
+This code results in the following output:
+
+```text
+This runtime coercion or type test from type
+    'a
+to
+    string
+involves an indeterminate type based on information prior to this program point. Runtime type tests are not allowed on some types. Further type annotations are needed.
+```
+
+In this code, the function parameter `x` has not had a type inferred for it. This could be for any number of reasons, ranging from the parameter not being used in ways that give the type-inference algorithm any information, to the function not being called anywhere else in the code.  Because the type of `x` has not been inferred, the compiler cannot guarantee that the type of `x` is able to be cast as a string.  Because the compiler cannot guarantee that this cast is able to be performed at runtime, it gives this message.
+
+To fix this message you must provide the compiler more information about the type of the value for which you are using the type-test pattern.
+
+One way of fixing the above code would be to apply a type to the `x` parameter directly:
+
+[!code-fsharp[FS0008-x-given-type](~/samples/snippets/fsharp/compiler-messages/fs0008.fs#8-11)]
+
+In this code the `x` parameter is given a type of `obj`, and since `obj` is able to be type-tested to see if the underlying representation is `string`, the function now compiles.
```

Comment:
```suggestion
In this code, the `x` parameter is given a type of `obj`, and since `obj` is able to be type-tested to see if the underlying representation is `string`, the function now compiles.
```